swingset/src/main/java/com/nqadmin/swingset/SSBaseComboBox.java

```diff
@@ -332,6 +401,143 @@ public boolean hasSelection() {
+		//      B. GlazedList bug where for the first character entered, the START_WITH matching is performed over the CONTAINS matching. In this case,
+		//         (getModel().getSize() > 0)
+		//         -> call super.setSelectedItem(getItemAt(0)) and continue
+		//      C. The user's entry into the editor results in zero matches. In this case ((getModel().getSize()==0) && (hasItems()==true))
+		//         -> set actionListenerNoUpdate to true so the Action Listener does not try to update the rowset
+		//         -> warn user, (attempt to) revert the editor string, and continue (simulated STRICT)
+	    //      D. SOMETHING ELSE???
+		//         -> set actionListenerNoUpdate to true so the Action Listener does not try to update the rowset
+		//         -> warn user and continue
+		//         
+		//  2. We get a valid SSListItem including nullItem if getAllowNull()==true. This could be from the UI or from a 
+		//     direct/indirect non UI call to setSelectedItem().
+		//	   -> call super.setSelectedItem(_value) and continue
+		//
+		//  3. We get a String or something else.
+		//     -> set actionListenerNoUpdate to true so the Action Listener does not try to update the rowset
+		//     -> warn user and return
+	
+		// INITIALIZATION
+		glGlitchItem = null; // WORKAROUND FOR GL 'CONTAINS' GLITCH
+		actionListenerNoUpdate = false; // TRUE WHEN THE USER HAS ENTERED SOMETHING RESULTING IN NO MATCHES
+
+		// GET CURRENT EDITOR TEXT IF AVAILABLE
+		final String currentEditorText;
+		if (getEditor().getItem() == null) {
+			currentEditorText = "";
+		} else {
+			currentEditorText = getEditor().getItem().toString();
+		}
+		
+		// INITIAL LOGGING
+		logger.debug(() -> String.format("%s: CALL TO setSelectedItem(%s), allowNull: %b, priorEditorText: '%s', currentEditorText: '%s'",
+				getColumnForLog(), _value, getAllowNull(), priorEditorText, currentEditorText));
+		
+		// EVALUATE SCENARIOS DOCUMENTED ABOVE
+		if (_value==null) {
+		// #1 - WE HAVE A NULL
+			if (!getEditor().getEditorComponent().hasFocus()) {
+	        // SCENARIO #1-A ABOVE - setting null from a rowset listener or programmatically (e.g., a default) 
+	        	super.setSelectedItem(null);
+	        	priorEditorText = currentEditorText;
+	        	logger.debug("{}: Null not from UI (e.g., rowset, default). Selected item after super.setSelectedItem()={}", () -> getColumnForLog(), () -> getSelectedItem());
+	        	
+			} else if (getModel().getSize() > 0) {
+			// SCENARIO #1-B ABOVE - null due to GlazedList glitch when items contain the first character, but GL does not match the first item returned
+				glGlitchItem = getItemAt(0);
+				super.setSelectedItem(glGlitchItem);
+				priorEditorText = getEditor().getItem().toString();
+				logger.debug("{}: Null due to GlazedLists glitch so selecting first item in model. Selected item after super.setSelectedItem()={}", () -> getColumnForLog(), () -> getSelectedItem());
+
+	        } else if (!currentEditorText.isEmpty()) {
```

Comment:
I'll turn on the space/tab display. Some testing indicates that 4 is the default for Eclipse which is odd because digging a bit indicates that tabs should be set to 8 for Java. Explains why I haven't noticed before as I've been using Eclipse for at least a decade.

Reply:
I committed a change to clean up two of the methods, but now thinking this may break the GitHub comments.